client/src/utils/socket.js

```diff
@@ -1,18 +1,30 @@
 import io from "socket.io-client";
-// import axios from "axios";
+import axios from "axios";
 
 class Socket {
-  async connect(token) {
+  connect(token, setNewNotification) {
```

Comment:
I'm worried a bit about the scalability of this method, since as we add more socket events, for example, when we add messages being received, we will need to update the messages component, and so I don't think we can feasibly add another method here to update messages. Do you know what I mean?

I think it's totally okay for now, but you'll need to rethink this in #61

Reply:
thanks for the feedback! will think about this and fix for next PR